File: monsoon3.0/src/main/java/wtf/monsoon/api/util/misc/ServerUtil.java
package wtf.monsoon.api.util.misc;

import net.minecraft.client.Minecraft;
import net.minecraft.client.multiplayer.ServerData;
import wtf.monsoon.api.util.Util;

import java.util.HashMap;
import java.util.Map;

public final class ServerUtil extends Util {

    public static boolean isOnServer(String ip) {
        if (mc.isSingleplayer())
            return false;

        return getCurrentServerIP().endsWith(ip);
    }

    public static String getCurrentServerIP() {
        if (mc.isSingleplayer())
            return "Singleplayer";

        return mc.getCurrentServerData().serverIP;
    }

    public static boolean isHypixel() {
        return isOnServer("hypixel.net") || isOnServer("ilovecatgirls.xyz");
    }

    public static String getCurrentServerIP(ServerData serverData) {
        if (mc.isSingleplayer())
            return "Singleplayer";

        return serverData.serverIP;
    }

    public static boolean isOnServer(String ip, ServerData serverData) {
        if (mc.isSingleplayer())
            return false;

        return getCurrentServerIP(serverData).endsWith(ip);
    }

    public static boolean isHypixel(ServerData serverData) {
        return isOnServer("hypixel.net", serverData) || isOnServer("ilovecatgirls.xyz", serverData);
    }
}

File: monsoon-3.x-v2/src/main/java/wtf/monsoon/client/util/network/ServerUtil.kt
package wtf.monsoon.client.util.network

import net.minecraft.client.multiplayer.ServerData
import wtf.monsoon.client.util.Util

object ServerUtil : Util() {

    fun isOnServer(ip: String): Boolean {
        return if (mc.isSingleplayer) false else getCurrentServerIP().endsWith(ip)
    }

    fun isConnectedToAnyServer(): Boolean {
        return if (mc.isSingleplayer) false else mc.currentServerData != null
    }

    fun getCurrentServerIP(): String {
        return if (mc.isSingleplayer) "Singleplayer" else mc.currentServerData.serverIP
    }

    fun isHypixel(): Boolean {
        return isOnServer("hypixel.net") || isOnServer("ilovecatgirls.xyz")
    }

    fun getCurrentServerIP(serverData: ServerData): String {
        return if (mc.isSingleplayer) "Singleplayer" else serverData.serverIP
    }

    fun isOnServer(ip: String?, serverData: ServerData): Boolean {
        return if (mc.isSingleplayer) false else getCurrentServerIP(serverData).endsWith(ip!!)
    }

    fun isHypixel(serverData: ServerData): Boolean {
        return isOnServer("hypixel.net", serverData) || isOnServer("ilovecatgirls.xyz", serverData)
    }

}